detox/local-cli/templates/jest.js

```diff
@@ -1,18 +1,27 @@
 const firstTestContent = require('./firstTestContent');
 const runnerConfig = `{
     "setupFilesAfterEnv": ["./init.js"],
-    "testEnvironment": "node"
-}`;
+    "testEnvironment": "node",
+    "reporters": ["detox/runners/jest/DetoxJestReporter.js"],
+    "verbose": true
+}
+`;
 
-const initjsContent = `const detox = require('detox');
-const config = require('../package.json').detox;
+const initjsContent = `const config = require('../package.json').detox;
 const adapter = require('detox/runners/jest/adapter');
+const traceAdapter = require('detox/runners/jest/traceAdapter');
 
+// Set the default timeout
 jest.setTimeout(120000);
 jasmine.getEnv().addReporter(adapter);
 
+// This takes care of generating status logs on a per-test basis.
+// By default, jest only reports at file-level.
+// This is strictly optional.
+jasmine.getEnv().addReporter(traceAdapter);
+
 beforeAll(async () => {
-  await detox.init(config);
+  await adapter.beforeAll(config);
```

Comment:
I've checked that explicitly using detox instance is a bad idea:

```
await adapter.beforeEach(detox);
await adapter.afterEach(detox);
```

That's because the adapter is flushing promises there with a reference to `this.detox` instance.